content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,235 @@
+is simply a kernel flag that controls what processes running within that namespace are able to see. For instance, if you
+create a new Linux network namespace via the `ip netns add my-linux-netns` command and run a process inside it, that process can only see the networking rules created
+within that network namespace. It can not see any network rules created outside of it - even though everything running on that machine is still sharing one Linux networking stack.
+
+Linux namespaces are conceptually a lot like Kubernetes namespaces - logical labels that organize and isolate different
+active processes, and allow you to create rules about what things within a given namespace can see and what rules are
+applied to them - they simply operate at a much lower level.
+
+When a process running within a network namespace creates a TCP packet outward bound for something else, the packet must be
+processed by any local rules within the local network namespace first, then leave the local network namespace, passing
+into another one.
+
+For example, in plain Kubernetes without any mesh installed, a pod might create a packet and send it to another pod, and
+the packet might (depending on how networking was set up):
+- Be processed by any rules within the source pod’s network namespace.
+- Leave the source pod network namespace, and bubble up into the node’s network namespace where it is processed by any rules in that namespace.
+- From there, finally be redirected into the target pod’s network namespace (and processed by any rules there).
+
+In Kubernetes, the [Container *Runtime* Interface (CRI)](https://kubernetes.io/docs/concepts/architecture/cri/) is responsible for talking to the Linux kernel, creating network namespaces
+for new pods, and starting processes within them. The CRI then invokes the [Container *Networking* Interface (CNI)](https://github.com/containernetworking/cni),
+which is responsible for wiring up the networking rules in the various Linux network namespaces, so that packets leaving and
+entering the new pod can get where they’re supposed to go. It doesn’t matter much to Kubernetes or the container runtime what topology or mechanism the CNI uses to accomplish this - as long as packets get where they’re supposed to be, Kubernetes works and everyone is happy.
+
+### Why did we drop the previous model?
+
+In Istio ambient mesh, every node has a minimum of two containers running as Kubernetes DaemonSets:
+- An efficient ztunnel which handles mesh traffic proxying duties, and L4 policy enforcement.
+- A CNI node agent that handles adding new and existing pods into the ambient mesh.
+
+In the previous ambient mesh implementation, this is how application pod is added to the ambient mesh:
+- The `istio-cni` node agent detects an existing or newly-started Kubernetes pod with its namespace labeled with `istio.io/dataplane-mode=enabled`, indicating that it should be included in the ambient mesh.
```

Comment:
```suggestion
- The `istio-cni` node agent detects an existing or newly-started Kubernetes pod with its namespace labeled with `istio.io/dataplane-mode=ambient`, indicating that it should be included in the ambient mesh.
```

Reply:
good catch!

Reply:
Same error in 2 places 😃 https://github.com/istio/istio.io/pull/14524#pullrequestreview-1849524995